File: 934-bitwise-ors-of-subarrays/bitwise-ors-of-subarrays.cpp
class Solution {
public:
    int subarrayBitwiseORs(vector<int>& arr) {
    
        unordered_set<int> prev;
        unordered_set<int> res;
        for(int i=0; i<arr.size(); i++) {    unordered_set<int> curr;
            for(auto &it: prev) {
                  curr.insert(arr[i] | it);
                  res.insert(arr[i] | it);
            }
            curr.insert(arr[i]);
            res.insert(arr[i]);
            prev=curr;
        }
        return res.size();
    }
};

File: 934-bitwise-ors-of-subarrays/bitwise-ors-of-subarrays.js
/**
 * @param {number[]} arr
 * @return {number}
 */
var subarrayBitwiseORs = function(arr) {
        let prev=new Set();
        let res= new Set();
        for(let i=0; i<arr.length; i++) {
            let curr=new Set();
            for(let val of prev) {
                curr.add(arr[i] | val);
                res.add(arr[i] | val);
            }

            curr.add(arr[i]);
            res.add(arr[i]);

            prev=curr;
        }

        return res.size;
};